### DISCUSSION TITLE
How to use Leva theme/styles in a custom plugin?

### DISCUSSION THREAD

**Discussion by Sec-ant** [pmndrs/leva#312] [Q&A] [answered]
I need a control that displays an identicon generated from some value. The most similar input control that Leva offers is the [Image Component](https://leva.pmnd.rs/?path=/story/inputs-image--image), but I don't need the drop zone. So I tried to write a custom plugin refering to the code in [Image Component](https://github.com/pmndrs/leva/blob/main/packages/leva/src/components/Image/Image.tsx). However the latter uses several styled components like `ImageContainer`, `ImagePreview`, `ImageLargePreview` that seem to be not exported. And I don't know how to use the styles from Leva themes like `rowHeight`, `colGap`, `sm`, etc. in my custom plugin. So by now I just hardcoded these styles with default values. My question is, how to use these styles in my custom plugin so that I won't have to hardcode these styles?

A simple piece of example code of my custom plugin (hardcoded styles):

[![Edit leva-custom-plugin (forked)](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/leva-custom-plugin-forked-bpff6o?fontsize=14&hidenavigation=1&theme=dark)

 Any help is appreciated, thanks!

**Comment by dbismut** [comment, marked as answer]
Have a look at the file below:

https://github.com/pmndrs/leva/blob/bdbe2332935a68bf4f26ae9f3efa5bcada7e3695/packages/plugin-bezier/src/StyledBezier.ts#L1

This comes from a plugin that inherits from the Leva panel styles.

**Reply by Sec-ant** [reply]
Thanks!

**Reply by Sec-ant** [reply]
@dbismut One more question, can I reference props in the `styled` function?

**Reply by dbismut** [reply]
This is a stitches specific question. You can using variants, this is explained in their docs.

**Reply by Sec-ant** [reply]
> This is a stitches specific question. You can using variants, this is explained in their docs.

Wow, this is quick. Haven't heard about stiches before, thanks for the explanation, I'll check it.

**Reply by dbismut** [reply]
You can thank GitHub notifications 😂 you can have a look at Leva's own source code to see how variants are being used. Stitches is the best styling lib IMO, near zero runtime + theming + variables is an incredible combo.
